lib/api/apiUtils/object/createAndStoreObject.js

```diff
@@ -273,6 +268,9 @@ function createAndStoreObject(bucketName, bucketMD, objectKey, objMD, authInfo,
             if (isPutVersion) {
                 const options = overwritingVersioning(objMD, metadataStoreParams);
                 return process.nextTick(() => next(null, options, infoArr));
+            } else if (!bucketMD.isVersioningEnabled() && objMD?.archive?.archiveInfo) {
```

Comment:
Maybe a good candidate for a unit test... It's a very sensitive function

Reply:
not sure unit test is the right place, but definitely a gap here → I created CLDSRV-561